lldb/source/Plugins/LanguageRuntime/Swift/LLDBMemoryReader.cpp

```diff
@@ -115,6 +117,41 @@ LLDBMemoryReader::getSymbolAddress(const std::string &name) {
   return swift::remote::RemoteAddress(load_addr);
 }
 
+swift::remote::RemoteAbsolutePointer
+LLDBMemoryReader::resolvePointer(swift::remote::RemoteAddress address,
+                                 uint64_t readValue) {
+  auto isEffectivelyReadOnly = [](SectionSP section) {
+    if ((section->GetPermissions() & ePermissionsWritable) == 0)
+      return true;
+    if (section->GetName() == "__const")
+      return true;
+    if (auto segment = section->GetParent())
+      if (segment->GetName() == "__DATA_CONST")
```

Comment:
We shouldn't be hardcoding implementation details of the MachO format here, but I'm sure we can sink this into Target or somewhere more appropriate

Reply:
thanks I am the same page, I put this change together quickly primarily to test asynchronously.